### PR TITLE
Add proper notifications for all tenant states

### PR DESCRIPTION
**Pull request by hishamco**
This is useful after we add Bulk Actions drop down

### CODE REVIEW

**Comment by sebastienros** [comment]
Sorry but you are adding new issues with the localization, just to handle cases that can't happen. For instance a tenant cannot be changed to "uninitilized" state.

**Comment by sebastienros** [comment]
You can close this PR

**Comment by hishamco** [comment]
> Sorry but you are adding new issues with the localization

Where?!!

> just to handle cases that can't happen. For instance a tenant cannot be changed to "uninitilized" state.

I know, but what I did if the tenant is "uninitialized" a notification should show up

**Comment by sebastienros** [comment]
What I mean is that with your change the tenant state is not localized anymore, like it will show "enabled" in all languages. And also you are handling cases that can never happen, like switching state to `uninitialized`. So I don't understand what good this PR is supposed to do. Unless you also explicitly state what is the issue you are trying to solve, because it's not obvious.

**Comment by hishamco** [comment]
> What I mean is that with your change the tenant state is not localized anymore, like it will show "enabled" in all languages.

You 're right about the localization, seems I added the action accidentally as parameter 

>  And also you are handling cases that can never happen, like switching state to uninitialized

If you look closely to `TenantState.Uninitialized` case for instance it is not change to `Uninitialized`, but if the tenant is `Uninitialized` and you try to click enable or disable a wrong alert show up

The intend for this PR is to display success alert in the following cases:

| Tenant State  | Action | Alert Type |
|---|---|---|
| Running | Enable | Warning |
| Running | Disable | Success |
| Disabled | Enable | Success |
| Disabled | Disable | Warning |
| Uninitialized | Enable | Error |
| Uninitialized | Disable | Error |

Hope this clarify the things

File: src/OrchardCore.Modules/OrchardCore.Tenants/Controllers/AdminController.cs [hunk 1]
```diff
@@ -554,21 +554,24 @@ private bool IsDefaultShell()
 
         private async Task UpdateShellSettingsAsync(ShellSettings settings, string action)
         {
+            string message;
+
             switch (settings.State)
             {
                 case TenantState.Running:
-                    _notifier.Warning(H["The tenant '{0}' is already {1}.", settings.Name, action]);
+                    message = String.Concat("The tenant '{0}' is already ", action, ".");
```
Comment (sebastienros) [inline src/OrchardCore.Modules/OrchardCore.Tenants/Controllers/AdminController.cs]: You can't localize a dynamic string. Comment applies to all subsequent usages too

Comment (hishamco) [reply src/OrchardCore.Modules/OrchardCore.Tenants/Controllers/AdminController.cs]: Strange, I expect all string will be evaluated before the `message` it passed in the localizer

Comment (hishamco) [reply src/OrchardCore.Modules/OrchardCore.Tenants/Controllers/AdminController.cs]: Strange, I expect all string will be evaluated before the `message` it passed in the localizer

Comment (sebastienros) [reply src/OrchardCore.Modules/OrchardCore.Tenants/Controllers/AdminController.cs]: Sure, but the string is still dynamically created. And there is no way to extract the possible texts in pot files from the localization extractor. 
Good: `H["some string {0}"]
Bad: `H[variable]`

**Comment by hishamco** [comment]
Just fix everything, we can refractor and bring `UpdateShellSettingsAsync` back when we support dynamic string localization

**Comment by sebastienros** [comment]
> when we support dynamic string localization

It has nothing to do with that. Dynamic string localization is about localizing strings that are defined by the user. Not strings that are computed dynamically like with concatenation or formatting.

**Comment by hishamco** [comment]
@agriffard could please have a look

**Comment by hishamco** [comment]
@jtkech could you please let me know if I did something wrong on this PR whenever it's suitable for you

**Comment by hishamco** [comment]
@agriffard could please have a look

**Comment by hishamco** [comment]
This PR is handling all the cases as shown in https://github.com/OrchardCMS/OrchardCore/pull/4169#issuecomment-529560300, we can ignore those case that supposed not to be happen

File: src/OrchardCore.Modules/OrchardCore.Tenants/Controllers/AdminController.cs [hunk 2]
```diff
@@ -167,47 +167,59 @@ public async Task<IActionResult> Index(BulkActionViewModel model)
...
-                        if (string.Equals(shellSettings.Name, ShellHelper.DefaultShellName, StringComparison.OrdinalIgnoreCase))
-                        {
-                            _notifier.Warning(H["You cannot disable the default tenant."]);
-                        }
-                        else if (shellSettings.State != TenantState.Running)
+                    case BulkAction.Disable:
+                        if (String.Equals(shellSettings.Name, ShellHelper.DefaultShellName, StringComparison.OrdinalIgnoreCase))
                         {
-                            _notifier.Warning(H["The tenant '{0}' is already disabled.", shellSettings.Name]);
+                            _notifier.Warning(H["You can't disable the default tenant."]);
```
Comment (Skrypt) [inline src/OrchardCore.Modules/OrchardCore.Tenants/Controllers/AdminController.cs:185]: Disabling the default tenant is not allowed.

**Comment by hishamco** [comment]
If we close this we need to close an issue related to this PR which I forgot to include

**Pull request closed without merging**